### DISCUSSION TITLE
Using RustPython to run HuggingFace dataset scripts

### DISCUSSION THREAD

**Discussion by antimora** [RustPython/RustPython#5379] [General]
Hello RustPython community,

I'm exploring the possibility of using RustPython to run HuggingFace dataset scripts as part of Burn, a deep learning framework in Rust. Our goal is to eliminate Burn's dependency on having Python installed. You can find more context about this task here: https://github.com/tracel-ai/burn/issues/2093

Specifically, I'm looking to achieve the following:

1. Use a minimal set of Python libraries and features necessary to run:
   - HuggingFace datasets library (https://github.com/huggingface/datasets)
   - HuggingFace dataset scripts (https://huggingface.co/docs/datasets/en/dataset_script)

2. Integrate this functionality into our Rust-based framework.

My questions are:

1. Is it feasible to use RustPython for this specific use case?
2. What would be the minimal set of Python features and libraries needed to support HuggingFace datasets?
3. Are there any potential limitations or challenges we should be aware of when using RustPython for this purpose?
4. Do you have any recommendations or best practices for integrating RustPython into a Rust project like ours?

Any insights or guidance would be greatly appreciated. Thank you in advance for your help!

**Comment by youknowone** [comment]
Trying it would be worth, but it will not be easy. Any library using C-API is not be supported

**Reply by antimora** [reply]
Oh, thanks for the c-api reminder. So I looked up datasets dependency and it appears it requires a bunch of c libs =(

```
[burn]% pip show datasets
Name: datasets
Version: 2.20.0
Summary: HuggingFace community-driven open-source library of datasets
Home-page: https://github.com/huggingface/datasets
Author: HuggingFace Inc.
Author-email: thomas@huggingface.co
License: Apache 2.0
Location: /opt/homebrew/lib/python3.12/site-packages
Requires: aiohttp, dill, filelock, fsspec, huggingface-hub, 
multiprocess, numpy, packaging, pandas, pyarrow, pyarrow-hotfix, pyyaml, requests, tqdm, xxhash
```

numpy, pandas and pyarrow are heavy users of c code.
